Mathlib/Data/Polynomial/Module.lean

```diff
@@ -350,3 +351,62 @@ theorem comp_smul (p p' : R[X]) (q : PolynomialModule R M) :
 #align polynomial_module.comp_smul PolynomialModule.comp_smul
 
 end PolynomialModule
+
+end CommSemiring
+
+namespace Module
+
+open Polynomial
+/--
+Suppose `a` is an element of an `R`-algebra `A` and `M` is an `A`-module.
+Then `Module.AEval R M a` is the `R[X]`-module with carrier `M`,
+where the action of `f : R[X]` is `f • m = (aeval a f) • m`.
+-/
+@[nolint unusedArguments]
+def AEval (R M: Type*) {A : Type*} [CommSemiring R] [Semiring A] [Algebra R A]
+    [AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M] (_ : A) := M
+
+variable {R A M} [CommSemiring R] [Semiring A] (a : A) [Algebra R A] [AddCommMonoid M] [Module A M]
+  [Module R M] [IsScalarTower R A M]
+
+namespace AEval
+
+instance : AddCommMonoid <| AEval R M a     := by assumption
+instance : Module R <| AEval R M a          := by assumption
+instance : Module A <| AEval R M a          := by assumption
+instance : IsScalarTower R A <| AEval R M a := by assumption
```

Comment:
It took me a minute to work out how these proofs even work, I think using `inferInstanceAs (AddCommMonoid M)` is probably cleaner